server/router/services/domain-service/helpers/fetch-domain-list-next-page.js

```diff
@@ -19,20 +19,46 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-const { DOMAIN_LIST_DELAY_MS, DOMAIN_LIST_PAGE_SIZE } = require('../constants');
+const {
+  DOMAIN_LIST_DELAY_MS,
+  DOMAIN_LIST_PAGE_SIZE,
+  DOMAIN_LIST_RETRY_MAX,
+} = require('../constants');
 const { delay } = require('../../../helpers');
 
 const fetchDomainListNextPage = async ({
   ctx,
   domainList = [],
   nextPageToken = '',
+  retryCount = 0,
 }) => {
-  const data = await ctx.cadence.listDomains({
-    pageSize: DOMAIN_LIST_PAGE_SIZE,
-    nextPageToken: nextPageToken
-      ? Buffer.from(encodeURIComponent(nextPageToken), 'base64')
-      : undefined,
-  });
+  let data;
+
+  if (retryCount >= DOMAIN_LIST_RETRY_MAX) {
+    return domainList;
+  }
+
+  try {
+    data = await ctx.cadence.listDomains({
+      pageSize: DOMAIN_LIST_PAGE_SIZE,
+      nextPageToken: nextPageToken
+        ? Buffer.from(encodeURIComponent(nextPageToken), 'base64')
+        : undefined,
+    });
+  } catch (error) {
+    console.error(
+      `fetchDomainListNextPage retry: ${retryCount} error: ${error.toString()}`
```

Comment:
Do we have a collection to read these failure messages later?

Reply:
not currently, but these will be available to our logging system and searchable.